MekHQ/src/mekhq/gui/view/PersonViewPanel.java

```diff
@@ -227,9 +297,100 @@ private void initComponents() {
         gridBagConstraints.fill = GridBagConstraints.BOTH;
         gridBagConstraints.anchor = GridBagConstraints.NORTHWEST;
         add(txtFiller, gridBagConstraints);
+    }
+
+    private void setRibbons() {
+    	int maxRibbonsPerRow = 4;
```

Comment:
This should at least be declared as a constant at the beginning of the class definition to make it easier to change. It would probably be better as a value in CampaignOptions, but I think user configuration of the layout might better be left for a followup.